wled00/FX_fcn.cpp

```diff
@@ -577,27 +577,23 @@ void Segment::setMode(uint8_t fx, bool loadDefaults) {
       sOpt = extractModeDefaults(fx, "mi");  if (sOpt >= 0) mirror    = (bool)sOpt; // NOTE: setting this option is a risky business
       sOpt = extractModeDefaults(fx, "rY");  if (sOpt >= 0) reverse_y = (bool)sOpt;
       sOpt = extractModeDefaults(fx, "mY");  if (sOpt >= 0) mirror_y  = (bool)sOpt; // NOTE: setting this option is a risky business
-      sOpt = extractModeDefaults(fx, "pal"); if (sOpt < 0) sOpt = 0; setPalette(sOpt, true); //else setPalette(0);
+      sOpt = extractModeDefaults(fx, "pal"); if (sOpt >= 0) setPalette(sOpt); //else setPalette(0);
     }
+    sOpt = extractModeDefaults(fx, "pal"); // always extract 'pal' to set _default_palette
+    if(sOpt == 0 || sOpt == 255) sOpt = 6; // partycolors if zero or not set (=255)
```

Comment:
255 is not "undefined" but is used for custom palettes. Custom palettes start at 255 and grow downwards.

Reply:
correct, it should be `-1` , this is a leftover from when I had this call inside setPalette() where it gets cast into an `uint8_t` resulting in 255